src/uniqueName.spec.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import {render} from 'react-dom'
+import {equal, ok} from 'assert'
+import uniqueName from './uniqueName'
+
+describe('uniqueName', () => {
+  describe('no name specified', () => {
+    it('has unique id', () => {
```

Comment:
You're not testing that it has a unique id, just that it has a name that may or may not be unique. Render a bunch of them and check that there are no collisions.